doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,132 @@
+application to embedded hardware.  This project template can be adapted to a new
+project, or its pieces can be incorporated into an existing CMake project.
+
+The first step is to clone the examples repository:
+
+```sh
+git clone git@github.com:mlpack/examples.git
+```
+
+Next, let's look at the `CMakeLists.txt` (e.g. the CMake configuration) in the
+`embedded/crosscompile_random_forest/` directory.
+
+### `CMakeLists.txt`
+
+The first part of the code, printed below or
+[available here](https://github.com/mlpack/examples/blob/master/embedded/crosscompile_random_forest/CMakeLists.txt),
+defines the project name and includes two useful CMake configuration files:
+ * `mlpack.cmake`: finds mlpack's dependencies and download them if necessary.
+ * `ConfigureCrossCompile.cmake`: set up CMake configuration for cross-compilation.
+ * `crosscompile-toolchain.cmake`: invoke CMake crosscompilation infrastructure.
+ * `crosscompile-arch-config.cmake`: add necessary flags depending on the
+   architecture (optional).
+
+Then we need to call `fetch_mlpack()` will download mlpack including all dependencies,
+cross-compile openblas and set up all the necessary parameters to find these dependencies.
+Most of mlpack dependencies are header-only with the exception of Openblas,
+thus this is expected to be a quick step.
+
+`fetch_mlpack()` will detect if cross compilation is necessary or not depending
+on the command that is executed when running cmake. Based on this, it will
+compile OpenBLAS.
+
+```cmake
+cmake_minimum_required(VERSION 3.11)
+project(main)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
+include(CMake/Autodownload.cmake)
+include(CMake/ConfigureCrossCompile.cmake)
+
+fetch_mlpack()
+
+```
+
+#### Setting up include directories and source files
+
+The last part of our CMake file consists of merging all of the above components
+together. First we are going to find the OpenMP package if the user defined the
+variable above, then we will start including the directories of our
+program. You will need to do the same for your software if you are trying to
+integrate this example into an existing development, this should be done by
+defining an include variable using `set` directive or add them directly in
+`target_include_directories`.
+Regarding the source code, a similar process by either adding then to
+`add_executables` or by appending the source files to `SOURCES_FILES` variable. 
+
+Finally do not forget to add any external library that you need to link against
+in `target_link_libraries`.
+
+```cmake
+include_directories(BEFORE ${MLPACK_INCLUDE_DIRS})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/src/)
+
+## User includes go here
+
+# Finally, add any cross-compilation support libraries (they may need to come
+# last).  If we are not cross-compiling, no changes will happen here.
+set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${CROSS_COMPILE_SUPPORT_LIBRARIES})
```

Comment:
Do you mean that `CROSS_COMPILE_SUPPORT_LIBRARIES` is meant to be set by the user, or is that something that is done by `ConfigureCrossCompile.cmake`?

Reply:
usually, it is done by `ConfigureCrossCompile.cmake`